Core/src/Seeding/CompSpacePointAuxiliaries.cpp

```diff
@@ -37,28 +43,67 @@ std::string StrawLineFitAuxiliaries::parName(const FitParIndex idx) {
   }
   return "unknown";
 }
-StrawLineFitAuxiliaries::StrawLineFitAuxiliaries(
+
+void CompSpacePointAuxiliaries::ChiSqWithDerivatives::reset() {
+  chi2 = 0.;
+  gradient.setZero();
+  hessian.setZero();
```

Comment:
an alternative would be to do 
```suggestion
  *this = ChiSqWithDerivatives();
```
or on the caller side instead of `.reset();` doing `= {};`

Reply:
Wouldn't that reallocate the memory?